daemon.go

```diff
@@ -64,28 +63,24 @@ func (d *menderDaemon) shouldStop() bool {
 func (d *menderDaemon) Run() error {
 	// set the first state transition
 	var toState State = d.mender.GetCurrentState()
+	var currentState State
 	cancelled := false
 	for {
-		// If signal SIGUSR1 received, check for update straight away.
+		// If signal SIGUSR1 or SIGUSR2 is received, force the state-machine to the correct state.
 		select {
-		case sig := <-d.updateCheck:
-			log.Debugf("received signal: %t", sig)
-			_, err := LoadStateData(d.sctx.store)
-			// No previous state stored, means no update was in progress,
-			// and we can safely force an update check.
-			if err != nil && os.IsNotExist(err) {
-				log.Debugf("received signal: %t. Forcing device from initState to updateCheck\n", toState)
-				if d.mender.IsAuthorized() {
-					toState = updateCheckState
-				} else {
-					toState = initState
-				}
-				d.mender.SetNextState(toState)
+		case nState := <-d.forceToState:
+			switch toState.(type) {
+			case *IdleState, *CheckWaitState:
+				log.Infof("Forcing state machine to: %s", nState)
+				toState = nState
 			}
+
 		default:
 			// Identity op - do nothing.
 		}
-		toState, cancelled = d.mender.TransitionState(toState, &d.sctx)
+
+		currentState = d.mender.TransitionState(toState, &d.sctx)
+		toState, cancelled = currentState.Handle(&d.sctx, d.mender)
```

Comment:
Moved the handling of the state out of the transition function. In my head this makes more sense. Tell me if you disagree.

Reply:
I kind of agree, but the update_modules branch, and in particular the big state test, relies pretty heavily on the old behavior. Maybe we can save this for another day?